vault/vault.go

```diff
@@ -192,6 +225,8 @@ func (t *tempCredsCreator) provider(config *Config) (credentials.Provider, error
 		}
 	} else if config.HasSSOStartURL() {
 		return NewSSORoleCredentialsProvider(t.keyring.Keyring, config)
+	} else if config.HasRole() && (config.HasWebIdentityTokenFile() || config.HasWebIdentityTokenProcess()) {
+		return NewAssumeRoleWithWebIdentityProvider(credentials.NewCredentials(sourceCredProvider), t.keyring.Keyring, config)
```

Comment:
Referencing `sourceCredProvider` inside this if-else block doesn't seem right, as the sourceCredProvider is defined in the if-else. Does this condition need to move after the final else?

Reply:
Good catch. Since we're using a web identity to assume a role, we don't actually need source credentials at all. This is a copy & paste error.

I've updated it in 6f25cc6 to use the same method for constructing a session as `NewSSORoleCredentialsProvider` which also does not require a `sourceCredProvider`.